The build has failed on OS X, but it has never passed before.

I've looked through the logs carefuly and noticed the wrong C++ compiler is used.
```
CMake Error at /nix/store/z9sf7kj1r3cj8pln31c6km3fc9g0dxmg-cmake-3.6.0/share/cmake-3.6/Modules/CMakeTestCXXCompiler.cmake:54 (message):
  The C++ compiler
  "/nix/store/f7l4pskmpnx57az1maz9v6ax38nb597j-clang-3.7.1/bin/clang++" is
  not able to compile a simple test program.
...
  /nix/store/f7l4pskmpnx57az1maz9v6ax38nb597j-clang-3.7.1/bin/clang++ -arch
  x86_64 -Wl,-search_paths_first -Wl,-headerpad_max_install_names
  CMakeFiles/cmTC_0e23b.dir/testCXXCompiler.cxx.o -o cmTC_0e23b
  ld: library not found for -lstdc++
  clang-3.7: error: linker command failed with exit code 1 (use -v to see
  invocation)
```

Unwrapped clang is used instead of wrapped one. Pushing the fix...

--- a/pkgs/development/tools/misc/ycmd/default.nix
+++ b/pkgs/development/tools/misc/ycmd/default.nix
@@ -10,12 +10,13 @@ stdenv.mkDerivation rec {
     sha256 = "1g0hivv3wla7z5dgnkcn3ny38p089pjfj36nx6k29zmprgmjinyr";
   };
 
-  buildInputs = [ python cmake llvmPackages.clang boost makeWrapper ];
+  buildInputs = [ python cmake boost makeWrapper ];
 
   propagatedBuildInputs = with pythonPackages; [ waitress frozendict bottle ];
 
   buildPhase = ''
-    python build.py --clang-completer --system-libclang --system-boost
+    export EXTRA_CMAKE_ARGS=-DPATH_TO_LLVM_ROOT=${llvmPackages.clang-unwrapped}
+    python build.py --clang-completer --system-boost
   '';
 
   configurePhase = ":";

